drivers/entropy/entropy_nrf5.c

```diff
@@ -272,44 +233,44 @@ static int entropy_nrf5_get_entropy_isr(struct device *dev, u8_t *buf, u16_t len
 	struct entropy_nrf5_dev_data *dev_data = DEV_DATA(dev);
 	u16_t cnt = len;
 
-	if (!(flags & ENTROPY_BUSYWAIT)) {
-		return get((struct rand *)dev_data->isr, len, buf);
+	if (likely((flags & ENTROPY_BUSYWAIT) == 0)) {
+		return rng_pool_get((struct rng_pool *)dev_data->isr, buf, len);
 	}
 
 	if (len) {
-		u32_t intenset;
+		unsigned int key;
+		int irq_enabled;
 
+		key = irq_lock();
+		irq_enabled = irq_is_enabled(RNG_IRQn);
 		irq_disable(RNG_IRQn);
-		NRF_RNG->EVENTS_VALRDY = 0;
-
-		intenset = NRF_RNG->INTENSET;
-		nrf_rng_int_enable(NRF_RNG_INT_VALRDY_MASK);
+		irq_unlock(key);
 
+		nrf_rng_event_clear(NRF_RNG_EVENT_VALRDY);
 		nrf_rng_task_trigger(NRF_RNG_TASK_START);
 
 		do {
-			while (NRF_RNG->EVENTS_VALRDY == 0) {
+			int byte;
+
+			while (!nrf_rng_event_get(NRF_RNG_EVENT_VALRDY)) {
```

Comment:
I don't entirely get this approach. Why for the flag `ENTROPY_BUSYWAIT` the pool is not used first, and the busy loop involved only if there is not enough random data available?
This flag is described in the API as "Busy-wait for random data to be ready". Does it imply that the random data should be generated in the busy loop (to be "fresh", not stored previously for some time)?

Reply:
This approach is simple and works and correctly in the case when request for random data comes from priority higher than RNG interrupt priority. Also, it leaves ISR pool untouched for the services which needs random data *fast*.

Reply:
It's probably worth a word of comment in the code then.